Bottom: 8f1d957bc9592a0114a23a7b456ed9e74bde2fd3
Top:    4e1a68eed06f029914ecf39c4895ea89be0cd2b3
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-02-13 18:57:51 +0000

polygon1.c: Fix unitialised output polygon upon when hitting allocation failure

This was a bug present in the poly_Boolean() routine. We need to set the output
polygon to NULL if we are going to return early on memory failure. This was
noticed during general review after observing problems due to our copy routines
incorrectly returning FALSE when copying a NULL polygon.


---

diff --git a/src/polygon1.c b/src/polygon1.c
index ed4c2bc..5cbf4ee 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -2388,6 +2388,8 @@ poly_Boolean (const POLYAREA * a_org, const POLYAREA * b_org,
 {
   POLYAREA *a = NULL, *b = NULL;
 
+  *res = NULL;
+
   if (!poly_M_Copy0 (&a, a_org) || !poly_M_Copy0 (&b, b_org))
     return err_no_memory;
